src/actions/wallet/L1/writeSendMessage.ts

```diff
@@ -0,0 +1,83 @@
+import { ResolveChain, WriteActionBaseType } from '../../../types/actions'
+import { OpStackL1Contracts } from '../../../types/opStackContracts'
+import { ContractToChainAddressMapping } from './writeUnsafeDepositTransaction'
+import {
+  l1CrossDomainMessengerABI,
+  l1StandardBridgeABI,
+} from '@eth-optimism/contracts-ts'
+import {
+  Account,
+  Address,
+  Chain,
+  Hex,
+  Transport,
+  WalletClient,
+  WriteContractParameters,
+  WriteContractReturnType,
+} from 'viem'
+import { writeContract } from 'viem/actions'
+
+export type SendMessageParams = {
+  target: Address
+  minGasLimit: number
+  message?: Hex
+}
+
+export type WriteSendMessageParameters<
+  TChain extends Chain | undefined = Chain,
+  TAccount extends Account | undefined = Account | undefined,
+  TChainOverride extends Chain | undefined = Chain | undefined,
+  _contractName extends OpStackL1Contracts = OpStackL1Contracts.optimismL1CrossDomainMessenger,
```

Comment:
This is an address not a name right?

Reply:
Seperate comment: this could be more typesafe about making people provide this if the contract isn't known on the chain but I would recomend adding that in a seperate pr

Reply:
typecheck should be good? Maybe pull and try it and lmk if not https://github.com/base-org/op-viem/blob/main/src/types/actions.ts#L18-L29

Reply:
Ah yea you are right lgtm